web3/utils/abi.py

```diff
@@ -438,11 +442,15 @@ def return_val_and_type(elements):
 
 @coerce_return_to_text
 def data_tree_vals(data_tree):
-    return recursive_map(lambda els: els[1] if _is_two_tuple(els) else els, data_tree)
+    values = recursive_map(lambda els: els[1] if _is_two_tuple(els) else els, data_tree)
```

Comment:
*nitpick* lambda is bugging me a bit.  Seems like a simple thing to

```
def _first_if_two_tuple(value):
    if is_to_tuple(value):
        return value[0]
    else:
        return value
```

which then makes the line in question:

```python
values = recursive_map(_first_if_two_tuple, data_tree)
```

which *might* be easier to read.... your call.

Reply:
Ok, attacked this, plus the issue of ambiguity (in case we end up with tuples of two values at some point, I don't want to accidentally trigger this mapping). Added a minimal class for type-checking. Next, to the docstrings...